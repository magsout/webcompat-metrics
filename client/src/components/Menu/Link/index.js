/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import PropTypes from "prop-types";
import { Link as LinkRouter } from "react-router-dom";

import classes from "./styles.css";

const Link = ({ to, text, icon }) => {
  return (
    <LinkRouter to={to}>
      <div className={classes.component}>
        {icon && <div className={classes.icon}>{icon}</div>}
        {text && <div className={classes.text}>{text}</div>}
      </div>
    </LinkRouter>
  );
};

Link.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  icon: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
};

Link.defaultProps = {
  text: null,
  icon: null,
};

export default Link;
