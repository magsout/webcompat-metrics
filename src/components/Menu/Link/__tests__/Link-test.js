/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";
import sinon from "sinon";

import SVGBugdiagnosis from "../../../../static/svg/Bugdiagnosis.svg";
import Link from "..";

it("renders Link default correctly", () => {
  const onClick = sinon.spy();
  const tree = renderer
    .create(
      <Link
        to="/weeklyreports"
        text={"Weekly Issue Reports"}
        icon={<SVGBugdiagnosis />}
        onClick={onClick}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
