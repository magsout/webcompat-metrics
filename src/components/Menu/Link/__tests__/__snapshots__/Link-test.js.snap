// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Link default correctly 1`] = `
<a
  href="/weeklyreports"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="component"
  >
    <div
      className="icon"
    >
      <svg
        viewBox="0 0 100 125"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          color="#000"
          d="M12 12.002c-3.29 0-6 2.71-6 6v50C6 71.29 8.71 74 12 74h29v10h-4.188c-1.047.05-1.955 1.047-1.906 2.094C34.956 87.14 35.953 88.05 37 88h26c1.057.015 2.028-.943 2.028-2s-.97-2.015-2.028-2h-4V74h29c3.29 0 6-2.71 6-6V18.003c0-3.29-2.71-6-6-6zm0 4h76c1.143 0 2 .857 2 2v24l-17.72-.032-6.468-13.812a2.048 2.048 0 0 0-1.874-1.18 2.05 2.05 0 0 0-1.813 1.274L51.187 55.97l-10.28-32.593c-.284-.893-1.26-1.507-2.188-1.375-.655.09-1.25.528-1.532 1.125L28.718 41H10V18.003c0-1.143.857-2 2-2zM38.625 29.47l10.47 33.124c.24.79 1.02 1.385 1.845 1.412a2.055 2.055 0 0 0 1.935-1.287l11.28-28.656 5.032 10.75A2.049 2.049 0 0 0 71 45.969L90 46v22c0 1.143-.857 2-2 2H12c-1.143 0-2-.857-2-2V45h20c.757 0 1.493-.47 1.813-1.155zM45 74h10v10H45z"
          overflow="visible"
          style={
            Object {
              "blockProgression": "tb",
              "textIndent": "0",
              "textTransform": "none",
            }
          }
        />
      </svg>
    </div>
    <div
      className="text"
    >
      Weekly Issue Reports
    </div>
  </div>
</a>
`;
